addons/mail/static/src/core/store_service.js

```diff
@@ -19,20 +20,47 @@ export class Store {
         this.isSmall = env.isSmall;
     }
 
+    /*
+     * Update the bus subscription if required. A subscription should be renewed
+     * if:
+     * - The user was added to the channel after the bus service started
+     * - The user left a channel
+     */
     updateBusSubscription() {
-        const channelIds = [];
+        const unknownChannelIds = [];
+        const knownChannelIds = new Set();
         const ids = Object.keys(this.threads).sort(); // Ensure channels processed in same order.
         for (const id of ids) {
             const thread = this.threads[id];
             if (thread.model === "discuss.channel" && thread.hasSelfAsMember) {
-                channelIds.push(id);
+                if (
+                    !thread.selfMember.memberSince ||
+                    thread.selfMember?.memberSince >= this.env.services["bus_service"].startedAt
+                ) {
+                    unknownChannelIds.push(id);
+                } else {
+                    knownChannelIds.add(id);
+                }
+            }
+        }
+        const nextChannelSubscription = JSON.stringify(unknownChannelIds);
+        let hasLeftChannel = this.lastKnownChannelIds.size > knownChannelIds.size;
+        if (!hasLeftChannel) {
+            for (const id of this.lastKnownChannelIds) {
+                if (!knownChannelIds.has(id)) {
+                    hasLeftChannel = true;
+                    break;
+                }
             }
         }
-        const channels = JSON.stringify(channelIds);
-        if (this.isMessagingReady && this.lastChannelSubscription !== channels) {
+        if (
+            this.isMessagingReady &&
+            (this.lastChannelSubscription !== nextChannelSubscription || hasLeftChannel)
+        ) {
             this.env.services["bus_service"].forceUpdateChannels();
         }
-        this.lastChannelSubscription = channels;
+        this.lastChannelSubscription = nextChannelSubscription;
+        this.lastKnownChannelIds = knownChannelIds;
```

Comment:
Shouldn't we update these only if we actually call `forceUpdateChannels`?

In particular, if `isMessagingReady` becomes ready later, we would miss all the intermediate ones?